src/MockClassifier.Api/Services/Dmr/DmrService.cs

```diff
@@ -11,73 +11,82 @@ namespace MockClassifier.Api.Services.Dmr
 
-            _httpClient = httpClientFactory.CreateClient(config.ClientName);
-            _httpClient.BaseAddress = config.DmrApiUri;
             _logger = logger;
-
-            requests = new ConcurrentQueue<DmrRequest>();
-
             _encodingService = encodingService;
         }
 
-        public void RecordRequest(DmrRequest request)
-        {
-            requests.Enqueue(request);
-        }
 
-        public async Task ProcessRequestsAsync()
+        public override async Task ProcessRequestAsync(DmrRequest payload)
         {
-            while (requests.TryDequeue(out var request))
+            if (payload == null || payload.Headers == null || payload.Payload == null)
+            {
+                throw new ArgumentNullException(nameof(payload));
+            }
+
+            try
             {
-                try
-                {
-                    // Setup content
-                    var jsonPayload = JsonSerializer.Serialize(request.Payload);
-                    var jsonPayloadBase64 = _encodingService.EncodeBase64(jsonPayload);
-                    using var content = new StringContent(jsonPayloadBase64, Encoding.UTF8, MediaTypeNames.Application.Json);
+                // Setup message
+                using var requestMessage = CreateRequestMessage(payload);
 
-                    // Setup message
-                    using var requestMessage = CreateRequestMessage(request, content);
+                // Send request
+                var response = await HttpClient.SendAsync(requestMessage).ConfigureAwait(false);
+                _ = response.EnsureSuccessStatusCode();
 
-                    // Send request
-                    var response = await _httpClient.SendAsync(requestMessage).ConfigureAwait(false);
-                    _ = response.EnsureSuccessStatusCode();
+                _logger.DmrCallback(payload.Payload.Classification, payload.Payload.Message);
+            }
+            catch (HttpRequestException exception)
+            {
+                _logger.DmrCallbackFailed(exception);
+            }
+        }
 
-                    _logger.DmrCallback(request.Payload.Classification, request.Payload.Message);
-                }
-                catch (HttpRequestException exception)
-                {
-                    _logger.DmrCallbackFailed(exception);
-                }
+        private static string EncodeBase64(string content)
+        {
+            if (content == null)
+            {
+                throw new ArgumentNullException(nameof(content));
             }
+
+            byte[] bytes = Encoding.UTF8.GetBytes(content);
+            var base64 = Convert.ToBase64String(bytes);
+            return base64;
         }
 
-        private static HttpRequestMessage CreateRequestMessage(DmrRequest request, StringContent content)
+        private static HttpRequestMessage CreateRequestMessage(DmrRequest request)
         {
```

Comment:
Would it be useful doing a null check here on the DmrRequest object's field 'request.payload'? that way we know we would not be sending an empty payload? forgive me if I have missed something here whilst making this point